packages/trader/src/AppV2/Components/TradeParameters/PayoutPerPoint/payout-per-point-wheel.tsx

```diff
@@ -1,87 +1,119 @@
+import { TTradeStore } from 'Types';
 
 type TPayoutPerPointWheelProps = {
     barrier?: string | number;
+    is_open?: boolean;
     current_payout_per_point: string;
     onPayoutPerPointSelect: (new_value: string | number) => void;
+    onClose: () => void;
     payout_per_point_list: {
         value: string;
     }[];
-    setV2ParamsInitialValues: ({ value, name }: { value: number | string; name: keyof TV2ParamsInitialValues }) => void;
 };
+type TOnProposalResponse = TTradeStore['onProposalResponse'];
 
-const onWheelPickerScrollDebounced = debounce(
-    (new_value: string | number, callback: TPayoutPerPointWheelProps['onPayoutPerPointSelect']) => callback(new_value),
-    200
-);
-
-const PayoutPerPointWheel = ({
-    barrier,
-    current_payout_per_point,
-    onPayoutPerPointSelect,
-    setV2ParamsInitialValues,
-    payout_per_point_list,
-}: TPayoutPerPointWheelProps) => {
-    const initial_value_ref = React.useRef<string | number>();
-    const selected_value_ref = React.useRef<string | number>(current_payout_per_point);
+const PayoutPerPointWheel = observer(
+    ({
+        barrier,
+        current_payout_per_point,
+        is_open,
+        onPayoutPerPointSelect,
+        onClose,
+        payout_per_point_list,
+    }: TPayoutPerPointWheelProps) => {
+        const trade_store = useTraderStore();
+        const { trade_types } = trade_store;
 
-    const onSave = () => {
-        initial_value_ref.current = selected_value_ref.current;
-        setV2ParamsInitialValues({ value: selected_value_ref.current, name: 'payout_per_point' });
-    };
+        const [value, setValue] = React.useState<string | number>(current_payout_per_point);
+        const [displayed_barrier_value, setDisplayedBarrierValue] = React.useState(barrier);
 
-    React.useEffect(() => {
-        if (!initial_value_ref.current && current_payout_per_point) {
-            initial_value_ref.current = current_payout_per_point;
-            setV2ParamsInitialValues({ value: current_payout_per_point, name: 'payout_per_point' });
-        }
+        // For handling cases when user clicks on Save btn before we got response from API
+        const is_api_response_received_ref = React.useRef(false);
 
-        return () => {
-            if (initial_value_ref.current && initial_value_ref.current !== selected_value_ref.current) {
-                onPayoutPerPointSelect(initial_value_ref.current);
+        const new_values = { payout_per_point: String(value) };
+        const proposal_req = getProposalRequestObject({
+            new_values,
+            trade_store,
+            trade_type: Object.keys(trade_types)[0],
+        });
+        // Sending proposal without subscription to get a new barrier value
+        const { data: response } = useDtraderQuery<Parameters<TOnProposalResponse>[0]>(
+            [
+                'proposal',
+                ...Object.entries(new_values).flat().join('-'),
+                `${barrier}`,
```

Comment:
Added barrier to the array with dependencies for cases when barrier from proposal was changed, but it didn't reflect on the value inside the modal